client/comms/wsconn.go

```diff
@@ -64,6 +65,20 @@ func (cs ConnectionStatus) String() string {
 	}
 }
 
+// invalidCertRegexp is a regexp that helps check for non-typed x509 errors
+// caused by or related to an invalid cert.
+var invalidCertRegexp = regexp.MustCompile(".*(unknown authority|not standards compliant|not trusted)")
+
+// IsErrorInvalidCert checks if the provided error is one of the different
+// variant of an invalid cert error returned from the x509 package or is
+// ErrInvalidCert.
+func IsErrorInvalidCert(err error) bool {
+	var invalidCert x509.CertificateInvalidError
+	var unknownCertAuth x509.UnknownAuthorityError
+	return errors.Is(err, ErrInvalidCert) || errors.Is(err, invalidCert) ||
+		errors.Is(err, unknownCertAuth) || invalidCertRegexp.MatchString(err.Error())
```

Comment:
This is what `errors.As` is for.  You don't need to create a dummy instance, and I'm not sure that would even work because you'd expect `err` to be an instance created elsewhere.